packages/p2p/src/components/app.jsx

```diff
@@ -69,124 +128,87 @@ const App = observer(props => {
         );
     }
 
+    const wrapper_props = { context_obj: getContext(), className };
+
+    if (general_store.show_popup) {
+        if (is_mobile) {
+            return (
+                <P2pWrapper {...wrapper_props}>
+                    <div className='p2p-nickname__dialog'>
+                        <NicknameForm
+                            handleClose={general_store.onNicknamePopupClose}
+                            handleConfirm={general_store.toggleNicknamePopup}
+                            is_mobile
+                        />
+                    </div>
+                </P2pWrapper>
+            );
+        }
+
+        return (
+            <P2pWrapper {...wrapper_props}>
+                <Modal is_open={general_store.show_popup} className='p2p-nickname__dialog'>
+                    <NicknameForm
+                        handleClose={general_store.onNicknamePopupClose}
+                        handleConfirm={general_store.toggleNicknamePopup}
+                    />
+                </Modal>
+            </P2pWrapper>
+        );
+    }
```

Comment:
Does this need to be two different returns when the only difference is in `is_open={general_store.show_popup}` and `is_mobile`? Can't they be managed using `isMobile`?

Reply:
They have a different wrapper too, i.e. `div` for `isMobile()` and `Modal` for the other case. I've added a condition in the `<P2pWrapper>` to avoid duplicating that, but I've left the two `<NicknameForm>`s to avoid another ugly HOC.